### DISCUSSION TITLE
200-million particle DEM column collapse

### DISCUSSION THREAD

**Discussion by jlinBE** [NVIDIA/warp#21] [Show and tell]
This is basically the column collapse example (example_dem.py), with the particle count bumped way up (500x500x800), exported into a flat file (skipping rendering) and then imported into a heavily modified version of the viewer from Nvidia Flex.  The viewer runs out of memory trying to display 200m particles, so it's subsampled to 25 million particles for visualization, with the (visual) particle radius doubled to compensate.  Colors represent velocity.

https://user-images.githubusercontent.com/51561503/162109521-af3404bc-b8cf-4353-863a-6ec521bc3d17.mp4
dropbox: https://www.dropbox.com/s/pc2ivsu2gwe33jo/output.mp4?dl=0

Around 9 hours on a single RTX-Titan - very fast for 200m particles!  Eyeballing the memory consumption with nvidia-smi, ~250m is probably the limit for 24GB of GPU memory.

**Comment by mmacklin** [comment]
Wow! Sorry I missed this, that's a big sim.!

Nice work.. I wonder how well USD would work for caching out the result / viewing in Omniverse.

Cheers,
Miles

**Comment by jlinBE** [comment]
Thanks!  For my project I just needed the raw particle positions, so I never bothered to get the Omniverse / USD parts working.
